### ISSUE TITLE
Rails/SaveBang: If you use a do block after a `create`... it won't recognize the `.persisted?` check.

### ISSUE THREAD

**Issue by arianf** [bbatsov/rubocop#3439]
https://github.com/bbatsov/rails-style-guide#save-bang

If you use a do block after a create... it won't recognize the `.persisted?` check.

--------

## Expected behavior
No error

## Actual behavior
Throws: 
> Rails/SaveBang: Use create! instead of create if the return value is not checked.  (https://github.com/bbatsov/rails-style-guide#save-bang)

## Steps to reproduce the problem

Turn on 

```yaml
Rails/SaveBang:
  Enabled: true
```

Then

```ruby
bob = User.create do |user|
  user.name = name
end

if bob.persisted?
  logger.error 'yay'
else
  logger.error 'meow'
end
```

## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.1.1 x86_64-darwin16.
```

**Comment by QuinnHarris** [comment]
The current cop does not do any more than verify that the result of a method is saved in a variable.  But this check does not work properly if a block was used as you show.
Actually checking if persisted? is checked is a good improvement though I expect to PR soon.
